flow/record/fieldtypes/__init__.py

```diff
@@ -609,6 +609,14 @@ def __new__(cls, *args):
 
         return cls._from_parts(args)
 
+    def __eq__(self, other: Any) -> bool:
+        if isinstance(other, str):
+            return str(self) == other
```

Comment:
I would also expect the following unit test to pass:

```python
from flow.record.fieldtypes import windows_path

def test_windows_path_eq():
    path = windows_path(r"c:\windows\test.exe")
    assert path == r"c:\windows\test.exe"
    assert path == r"c:/windows/test.exe"
    assert path == r"c:/windows\test.exe"
```

I think we could do fast pass str compare and otherwise initialize path (which is more expensive):
```suggestion
            return str(self) == other or self.__class__(other)
```